contrib/upgradability-patterns.asciidoc

```diff
@@ -0,0 +1,367 @@
+   * @param record sha3 key
+   */
+  function deleteIntValue(bytes32 record) public onlyOwner {
+    delete IntValues[record];
+  }
+
+  /**
+   * @dev change the value inside StringValues for the given key 
+   * @param record sha3 key
+   * @param value new value is string
+   */
+  function setStringValue(bytes32 record, string value) public onlyOwner {
+    StringValues[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside StringValues for the given key 
+   * @param record sha3 key
+   */
+  function deleteStringValue(bytes32 record) public onlyOwner {
+    delete StringValues[record];
+  }
+
+  /**
+   * @dev change the value inside AddressValues for the given key 
+   * @param record sha3 key
+   * @param value new value is address
+   */
+  function setAddressValue(bytes32 record, address value) public onlyOwner {
+    AddressValues[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside AddressValues for the given key 
+   * @param record sha3 key
+   */
+  function deleteAddressValue(bytes32 record) public onlyOwner {
+    delete AddressValues[record];
+  }
+
+  /**
+   * @dev change the value inside BytesValues for the given key 
+   * @param record sha3 key
+   * @param value new value is bytes
+   */
+  function setBytesValue(bytes32 record, bytes value) public onlyOwner {
+    BytesValues[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside BytesValues for the given key 
+   * @param record sha3 key
+   */
+  function deleteBytesValue(bytes32 record) public onlyOwner {
+    delete BytesValues[record];
+  }
+
+  /**
+   * @dev change the value inside Bytes32Values for the given key 
+   * @param record sha3 key
+   * @param value new value is bytes32
+   */
+  function setBytes32Value(bytes32 record, bytes32 value) public onlyOwner {
+    Bytes32Values[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside Bytes32Values for the given key 
+   * @param record sha3 key
+   */
+  function deleteBytes32Value(bytes32 record) public onlyOwner {
+    delete Bytes32Values[record];
+  }
+
+  /**
+   * @dev change the value inside BooleanValues for the given key 
+   * @param record sha3 key
+   * @param value new value is bool
+   */
+  function setBooleanValue(bytes32 record, bool value) public onlyOwner {
+    BooleanValues[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside BooleanValues for the given key 
+   * @param record sha3 key 
+   */
+  function deleteBooleanValue(bytes32 record) public onlyOwner {
+    delete BooleanValues[record];
+  }
+}
+----
+
+*CounterContractV1*
+----
+pragma solidity ^0.4.13;
+
+import "zeppelin-solidity/contracts/lifecycle/Pausable.sol";
+import "EternalStorage.sol";
+
+contract CounterWithEternalStorage is Pausable {
+
+  EternalStorage eternalStorage;
+
+  function CounterWithEternalStorage() {
+    eternalStorage = new EternalStorage();
+  }
+
+  function getCount() public constant returns(uint256) {
```

Comment:
Should this function `public`?

Reply:
Why not? All the eternalStorage functions that return data are public.